value.go

```diff
@@ -163,6 +165,20 @@ func (v Value) Copy() *Value {
 	return &Value{v.Type, b}
 }
 
+// CopyFrom copies from into v, reusing v.Bytes if possible and setting v.Bytes
+// to nil if and only if from.Bytes is nil.
+func (v *Value) CopyFrom(from *Value) {
+	v.Type = from.Type
+	if from.Bytes == nil {
+		v.Bytes = nil
+	} else if v.Bytes == nil {
+		v.Bytes = make(zcode.Bytes, len(from.Bytes))
+		copy(v.Bytes, from.Bytes)
+	} else {
+		v.Bytes = append(v.Bytes[:0], from.Bytes...)
+	}
+}
+
```

Comment:
I'm confident we want something like this, but I'm not sure I got the interface quite right.

Reply:
That looks good for now but I think we want to package all this up inside of batch for most use cases so we can efficiently clone  values that share byte slices where they all live inside the same batch.